protocol/tests/fileio.rs

```diff
@@ -0,0 +1,44 @@
+//  Copyright (c) Facebook, Inc. and its affiliates.
+//  SPDX-License-Identifier: Apache-2.0
+
+extern crate protocol;
+use protocol::fileio::*;
+use std::{
+    collections::HashMap,
+    path::Path,
+    sync::{Arc, RwLock},
+};
+
+#[test]
+fn test_load_data() {
+    let t = KeyedCSV {
+        headers : vec![],
+        records : HashMap::from([
+            (String::from("a"), vec![String::from("1"), String::from("10")]),
+            (String::from("b"), vec![String::from("2"), String::from("20")]),
+            (String::from("c"), vec![String::from("3"), String::from("30")]),
+            (String::from("d"), vec![String::from("4"), String::from("40")]),
+        ]),
+    };
+
+
+    let plain_data = Arc::new(RwLock::new(KeyedCSV::default()));
```

Comment:
Curious why you need `c` - couldn't `plain_data.read().unwrap()` work in line 29?

Reply:
> plain_data

plain_data has been moved to function load_data